example/src/main/java/io/netty/example/http/upload/HttpUploadServerHandler.java

```diff
@@ -225,12 +225,8 @@ private void readHttpDataChunkByChunk() {
                         logger.info(" 100% (FinalSize: " + partialContent.length() + ")");
                         partialContent = null;
                     }
-                    try {
-                        // new value
-                        writeHttpData(data);
-                    } finally {
-                        data.release();
-                    }
+                    // new value
+                    writeHttpData(data);
```

Comment:
writeHttpData always guarantees to release data even if it throws an exception for some reason?

Reply:
writeHttpData just appends the content of chunks to StringBuilder it doesn't and shouldn't release data.

Actual release happens in reset() method.

Reply:
inside finally block?

Reply:
Why should it?

Reply:
Just asking if it invokes HttpPostRequestDecoder.destroy() in all cases to avoid memory leaks.

Reply:
HttpPostRequestDecoder.destroy() invoked in exceptionCaught and in channelInactive. Not the best places, however, this should work fine in case of error.

Reply:
aha ok!